package.json

```diff
@@ -9,8 +9,11 @@
     "test": "react-scripts test",
     "test:update": "npm test -- --watchAll=false --updateSnapshot",
     "test:ci": "npm test -- --watchAll=false --coverage --runInBand --no-cache",
-    "e2e:dev": "node -e 'require(\"./src/e2eConfig\").e2eRunDev(\"http://localhost:3000\")'",
-    "e2e": "PUBLIC_URL=http://localhost:3000 npm run build && node -e 'require(\"./src/e2eConfig\").e2eRun(\"3000\")'",
+    "e2e:dev": "concurrently 'npm run start' 'npm run cypress:dev'",
```

Comment:
Can't we just use `npm run start && npm run cypress:dev`? Don't we have a potential race condition here?

Reply:
from what I see, this would be a viable candidate to solve this issue:
https://www.npmjs.com/package/start-server-and-test

Reply:
Nope 🙂. Race condition is not an issue in this case. If you try to run `npm run start && npm run cypress:dev` the second task `npm run cypress:dev` will never be triggered. Simply because `npm start` command runs script `react-scripts start` which later starts webpack's dev server with livereload. That's why first process never ends (until you kill it of course) and second is never triggered.

Reply:
> from what I see, this would be a viable candidate to solve this issue:
> https://www.npmjs.com/package/start-server-and-test

Okay, you forced me to read README for this tool 🤣. The author *Gleb Bahmutov* (a good conference speaker)  also uses `wait-on` dependency under the hood. Just like i did. `concurrently` allows you to run in parallel any task you want and it's not tied to web server. Therefore it's more elastic IMO. 

>If you are using webpack-dev-server (directly or via angular/cli or other boilerplates) then please use the following URL form to check
>start-server-and-test http-get://localhost:8080
>This is because under the hood this module uses wait-on to ping the server. Wait-on uses HEAD by default, but webpack-dev-server does not respond to HEAD only to GET requests. Thus you need to use http-get:// URL format to force wait-on to use GET probe.

Source: https://www.npmjs.com/package/start-server-and-test